content/millennium/bulk-data/group-export.md

```diff
@@ -0,0 +1,273 @@
+
+Response
+
+    Status: 200 OK
+    Expires: Mon, 22 Jul 2025 23:59:59 GMT
+    Content-Type: application/json
+
+<%= json(:R4_BULK_EXPORT_COMPLETE) %>
+
+<%= disclaimer %>
+
+### Errors
+
+The common [errors] and [OperationOutcomes] may be returned.
+
+## File Retrieval Request
+
+Once an export is complete, the status endpoint will return file URLs which may be used to download the exported resources
+
+    GET /bulk-export/files/<file_id>
+
+_Implementation Notes_
+
+* The file retrieval endpoint returns a '307 Temporary Redirect' with a 'Location' header, the Location header must be followed in order to download the file
+* An exported ndjson file contains only a single resource type and there can be multiple files for the same resource type. There is no limit number of resources per file.
+* Files should be retrieved promptly, as they will expire after 30 days.
+* The initial file retrieval request requires the Authorization header to be sent, when following the re-direct, it is no longer appropriate. Ensure the Authorization header is not perpetuated.
+* The Authorization token must contain the system read scope for the resource that the export file contains.
+
+_Retrieval of Attachments_
+
+* Binary resources are not returned by bulk data operations.
+* Attachment references must be retrieved separately using the Binary resource URL from the DocumentReference.content.attachment and DiagnosticReport.presentedForm fields from the export files.
+
+### Authorization Types
+
+<%= authorization_types(provider: false, patient: false, system: true) %>
+
+### Headers
+
+<%= headers head: {Authorization: '&lt;OAuth2 Bearer Token>', Accept: 'application/fhir+ndjson'} %>
+
+### Example
+
+Request
+
+    GET https://fhir-ehr.cerner.com/r4/ec2458f2-1e24-41c8-b71b-0e701af7583d/bulk-export/files/<file_id>
+
+Response
+
+    Status: 307 Temporary Redirect
+    Location: https://cerner-opensvc-bulkfhir-processed.s3.us-west-2.amazonaws.com
+
+<%= disclaimer %>
+
+### Example: Following the 307 Redirect
+
+Request
+
+    GET https://cerner-opensvc-bulkfhir-processed.s3.us-west-2.amazonaws.com
+
+_Notes_
+
+* When following the re-direct link from the 'Location' header, an Authorization header should NOT be set as the url is pre-signed.
+* Re-direct links are short-lived.
+* Files are deleted 30 days after creation.
+
+Response
+
+    Status: 200 OK
+    Content-Type: application/fhir+ndjson
+
+<%= ndjson(:R4_BULK_EXPORT_FILE) %>
+
+<%= disclaimer %>
+
+### Errors
+
+The common [errors] and [OperationOutcomes] may be returned.
+
+## Delete Request
+
+Use the URL returned in the `Content-Location` header from the kick-off request to delete a bulk data export job (whether it's in-progress or complete) and any associated files:
+
+    DELETE /bulk-export/jobs/<job_id>
+
+### Authorization Types
+
+<%= authorization_types(provider: false, patient: false, system: true) %>
```

Comment:
What are the scopes?

Reply:
Added a note on the scopes here: https://github.com/cerner/fhir.cerner.com/pull/899/commits/eff91ed66335ec7340cb5a2ec8a51663c1bd9cbf#diff-c2a4803bbbf3b17e514d41cc6eb7d578c94a283f937cce0400f5437570744d7fR275-R278